testing/fstest/fstest_main.c

```diff
@@ -188,6 +189,29 @@ static inline char fstest_randchar(void)
     }
 }
 
+/****************************************************************************
+ * Name: fstest_checkexit
+ ****************************************************************************/
+
+static bool fstest_checkexit(FAR struct fstest_filedesc_s *file)
+{
+  int i;
+  bool ret = false;
+
+  for (i = 0; i < CONFIG_TESTING_FSTEST_MAXOPEN; i++)
+    {
+      if (!g_files[i].deleted &&
+          &g_files[i] != file &&
+          g_files[i].hash == file->hash)
```

Comment:
why not directly compare filename to avoid the potential collision?

Reply:
Using hash just want to speed up process when files and names are a little of more.